fast/stages/2-project-factory/README.md

```diff
@@ -0,0 +1,354 @@
+name: Team A
+# use the explicit id of the Teams folder
+parent: folders/1234567890
+```
+
+```yaml
+name: Team A
+# use variable substitution from stage 1 tfvars (preferred approach)
+parent: teams
+```
+
+The third way explained above does not explicitly define a root folder in the YAML files, but sets a default folder in the Terraform variables for the stage via the `factories_config.substitutions.folder_ids`, by adding a `default` key pointing to the folder id of the root ("Teams") folder.
+
+```tfvars
+factories_config = {
+  substitutions = {
+    folder_ids = {
+      # id of the top-level Teams folder
+      # derived from the 1-resman.auto.tfvars.json file
+      default = "folders/12345678"
+    }
+  }
+}
+# tftest skip
+```
+
+### Project Creation
+
+Project YAML files can be created in two different filesystem paths:
+
+- in the filesystem folder defined via the `factories_config.project_data` variable, and then explicitly setting their `parent` attribute in YAML files, or
+- in the filesystem hierarchy discussed above, so that their `parent` attribute is automatically derived from the containing folder
+
+The two approaches can be mixed and matched, but the first approach is safer as is avoids potentially dangerous situations when folders are deleted with project configuration files still inside.
+
+When specifying projects outside of the folder hierarchy, setting the parent folder works in pretty much the same way as discussed above, with substitutions available for any folder defined in the filesystem hierarchy. This allows writing portable files, by referring to short names instead of resource ids.
+
+```yaml
+# use the explicit id of the parent folder
+parent: folders/1234509876
+```
+
+```yaml
+# use variable substitution from managed folders (preferred approach)
+parent: team-a/dev
+```
+
+All of the [project module](../../../modules/project/) attributes (and some service account attributes) can of course be leveraged in the configuration files. Refer to the [project schema](./schemas/folder.schema.json) for the complete set of available attributes.
+
+### Automation Resources for Projects
+
+When created projects are meant to be managed via IaC downstream, an initial set of automation resources can be created in a "controlling project". The preferred pattern is to first create one or more controlling projects for the project factory, and then leverage them for service account and GCS bucket creation.
+
+```yaml
+# controlling project shown in the diagram above
+parent: teams
+name: xxx-prod-iac-teams-0
+services:
+  - compute.googleapis.com
+  - storage.googleapis.com
+  # ...
+  # enable all services used by service accounts in this project
+```
+
+Once a controlling project is in place, it can be used in any other project declaration to host service accounts and buckets for automation. The service accounts can be used in IAM bindings in the same file by referring to their name via substitutions, as shown here.
+
+```yaml
```

Comment:
this one I'd definitely validate against the schema

Reply:
![image](https://github.com/user-attachments/assets/af18b3d2-5af1-4094-a960-c40fd4558bb0)
 :)

Reply:
Standalone stanza does not seem to trigger a test. I'll leave it to you to figure it out :)

```
# tftest-file schema=project.schema.json
```

Reply:
standalone validation passes

```
2024/08/20 16:05:12 Loading schema modules/project-factory/schemas/project.schema.json...
2024/08/20 16:05:12 Validating /tmp/test.yaml...
2024/08/20 16:05:12 OK
```